source/hp/dof_handler.cc

```diff
@@ -227,7 +227,7 @@ namespace internal
           vertex_fe_association (dof_handler.finite_elements->size(),
                                  std::vector<bool> (dof_handler.tria->n_vertices(), false));
 
-          for (typename DoFHandler<dim,spacedim>::active_cell_iterator
+          for (TriaActiveIterator < dealii::DoFCellAccessor < DoFHandler < dim, spacedim >, false > >
```

Comment:
Just fyi the exact exception message I am getting when I leave the code from dealii repo untouched:

[dealii::StandardExceptions::ExcMessage] = {arg1="You are trying to assign iterators that are incompatible. Reasons for incompatibility are that they point to different types of DoFHandlers (e.g., dealii::DoFHandler and dealii::hp::DoFHandler) or tha... }